### PR TITLE
Switch to external library for getting image sizes

### PR DESCRIPTION
**Pull request by Bond-009**
I wrote ImageSize.Net from scratch to replace ImageHeader.cs which
supports only the most common versions of the image formats.
ImageSize.Net support more edge cases and has been heavily testing
(in contrast to ImageHeader.cs which was copied from SO)
Project link: https://github.com/Bond-009/ImageSize.Net

### CODE REVIEW

File: Emby.Drawing/ImageProcessor.cs [hunk 1]
```diff
@@ -196,7 +199,7 @@ public bool SupportsTransparency(string path)
 
             var originalImagePath = originalImage.Path;
             var dateModified = originalImage.DateModified;
-            var originalImageSize = originalImage.Width > 0 && originalImage.Height > 0 ? new ImageSize(originalImage.Width, originalImage.Height) : (ImageSize?)null;
```
Comment (hawken93) [inline Emby.Drawing/ImageProcessor.cs]: Can this code be refactored to not use the <test> ? <if> : <else> construct? It was hard to read before, and with the new addition it's even harder to read. :)

Comment (JustAMan) [reply Emby.Drawing/ImageProcessor.cs]: 👍

File: Emby.Drawing/ImageProcessor.cs [hunk 2]
```diff
@@ -395,7 +398,7 @@ private string GetMimeType(ImageFormat format, string path)
         /// <summary>
         /// Gets the cache file path based on a set of parameters
         /// </summary>
-        private string GetCacheFilePath(string originalPath, ImageSize outputSize, int quality, DateTime dateModified, ImageFormat format, bool addPlayedIndicator, double percentPlayed, int? unwatchedCount, int? blur, string backgroundColor, string foregroundLayer)
+        private string GetCacheFilePath(string originalPath, MediaBrowser.Model.Drawing.ImageSize outputSize, int quality, DateTime dateModified, ImageFormat format, bool addPlayedIndicator, double percentPlayed, int? unwatchedCount, int? blur, string backgroundColor, string foregroundLayer)
```
Comment (hawken93) [inline Emby.Drawing/ImageProcessor.cs]: It would also be nice to split the arguments on newlines so they are easier to read here :)

**Comment by nvllsvm** [comment]
@Bond-009 can you make it apart of the Jellyfin org? I don't want to Jellyfin to depend upon an external library with a short history and no apparent use.

**Pull request closed without merging**